### DISCUSSION TITLE
整合 md 和 ipynb 于一体

### DISCUSSION THREAD

**Discussion by xinetzone** [mlc-ai/mlc-zh#27] [Ideas]
为了方便做笔记，我建立了 @[daobook/mlbook](https://github.com/daobook/mlcbook)。它整合 md 和 ipynb 于一体。

![](https://user-images.githubusercontent.com/30319120/177232412-a5996fff-9d9c-412f-a680-0ec49fddc43c.png)
![](https://user-images.githubusercontent.com/30319120/177233126-702babe2-ec07-486a-a7ee-3d7aef747cc6.png)
![](https://user-images.githubusercontent.com/30319120/177233196-417673a4-6bd7-4157-87e6-3515d2c03fe2.png)

支持：

1. 在 Colab 运行
2. 直接在页面下载 PDF 和 notebook
3. 支持讨论
4. 使用 `invoke doc`  快速生成web版本文档

个人能力有限，仅仅实现这些功能。

不知道是否可以对本课程有所帮助。😊

**Comment by xinetzone** [comment]
![自动化部署到 gh-pages](https://user-images.githubusercontent.com/30319120/177234681-319c3c7c-d9e6-47b1-87b7-0e6951d6e583.png)

[.github/workflows/deploy-gh-pages.yml](https://github.com/daobook/mlcbook/blob/xin/.github/workflows/deploy-gh-pages.yml) 提供了快速创建 TVM 环境的方案。
![](https://user-images.githubusercontent.com/30319120/177234847-534e027d-dee8-43ba-ad07-47982cbda9f0.png)

其中 `inv doc` 用于生成 web 文档。

**Comment by hbsun2113** [comment]
请问如何使用，比如可以自己创建文件吗？
我无法通过Colab打开你的文件，显示没有权限。

**Reply by xinetzone** [reply]
😂抱歉。colab 需要设置分支：

![](https://user-images.githubusercontent.com/30319120/177981967-d247d0c5-20ca-4a34-b825-4a5a9470387b.png)

这里设定 `xin` ，是因为文档是 `xin`

**Reply by xinetzone** [reply]
可以自己创建新的笔记本
只要在 `index.md` 添加文件名即可。具体教程可以参考：[jupyter-book](https://daobook.github.io/jupyter-book-zh/)。
